libtransact/src/workload/mod.rs

```diff
@@ -48,6 +48,18 @@ pub trait BatchWorkload: Send {
         -> Result<(BatchPair, Option<ExpectedBatchResult>), InvalidStateError>;
 }
 
+/// `TransactionReader` provides an API for reading transactions from a source
+pub trait TransactionReader {
+    /// Returns a vec with at least max_txns `Transactions` in it
+    fn next(&mut self, max_txns: usize) -> Result<Vec<Transaction>, InvalidStateError>;
```

Comment:
These should return InternalError. InvalidStateError is primarily for things like builders that the user can get into an invalid state by calling inconsistent functions or insufficient functions prior to build().

Reply:
Fixed